### ISSUE TITLE
All leds are too bright

### ISSUE THREAD

**Issue by huksley** [adammhaile/RPi-LPD8806#12]
Hello!
Thanks for a great library!

I`ve bought 2m LPD8806 strip and trying to use it with RPI. Powering it off ATX PSU via molex connector.

When starting example.py all leds light up on maximum brightness.
Colors are changing, but chaotically, not smooth like in your library.
Here is the video:
http://www.youtube.com/watch?v=vYSRQso3hak

Do you have any suggestions on this strange behaviour?
Thanks in advance!

**Comment by adammhaile** [comment]
Wow... those are crazy bright. Where did you get the lights from?
Are you using a signal level converter? The chaotic changing seems to be a
signal problem.

**Comment by huksley** [comment]
I am not very proficient in electronics. 
Got it from adafruit, www.adafruit.com/products/306

I`ve just connected DI/CI to MOSI/SCLK directly, as recommended.
The only issue I observe myself is ATX PSU giving out 5.3 on molex without load.
Also I`ve got Raspberry PI from production batch which is known to have bad soldering resulting in overheating under load, but after baking it in oven, it is working ok.

**Comment by adammhaile** [comment]
Where was it recommended to hook up the signal lines directly?
Yes, in theory it works, but 3.3V is technically below the logic level
threshold for the LPD8806 chips.
You can hook them up directly ONLY if using a 5V microcontroller, but the
Rasp Pi is 3.3V.
I use one of these: http://www.adafruit.com/products/757

Also, 5.3V is too high, and probably why they are so bright. You won't burn
them out at that voltage, but it's sill not good.

**Comment by huksley** [comment]
Thanks for the link and help.
I will try to connect to Arduino to see if led strip is ok.
Getting adafruit level converter is difficult, the shipping will take months.
Maybe I can assemble it myself.

**Comment by grantemsley** [comment]
For what its worth I have the exact same LEDs from adafruit running off an
RPi.  I only have 1m connected, but it works perfectly at 5v for power and
3.3v logic from the RPi.

Have you tried lighting the LEDs up one at a time and seeing how it
reacts?  It could be the strand is too long and not getting enough power at
the far end, or the power supply can't provide enough amps to turn all the
LEDs on at once.

**Comment by huksley** [comment]
I avoided using direct ATX PSU 5v line and used instead LM2596 based stabilizer to get 12v -> 5v to power ledstrip.
Using such power supply and connected to arduino everything works (every led is controllable and not so bright)

However, using it in RPI gives same random result. 
I reinstalled Raspian from scratch,
and lowered hw spi frequency to 1Mhz, no difference.

Is there any way to debug SPI on RPI?
Thanks for help.
